services/QuillLMS/client/app/bundles/Evidence/components/studentView/container.tsx

```diff
@@ -200,7 +203,120 @@ export const StudentViewContainer = ({ dispatch, session, isTurk, location, acti
+    setShowReadTheDirectionsButton(false)
+    setScrolledToEndOfPassage(false)
+    setDoneHighlighting(false)
+    setHasStartedPromptsSteps(false)
+    setCompleteButtonClicked(false)
+    setCompletedSteps([])
+  }
+
+  function preparePreviewChecklistStep() {
+    dispatch(setPreviewSessionStep(CHECKLIST))
+    dispatch(setActiveStepForSession(1))
+    dispatch(setExplanationSlidesCompletedForSession(true))
+    dispatch(setActivityIsCompleteForSession(false))
+    setHasStartedReadPassageStep(false)
+    setShowReadTheDirectionsButton(false)
+    setScrolledToEndOfPassage(false)
+    setDoneHighlighting(false)
+    setHasStartedPromptsSteps(false)
+    setCompleteButtonClicked(false)
+    setCompletedSteps([])
+  }
+
+  function preparePreviewReadAndHighlightStep() {
+    dispatch(setPreviewSessionStep(READ_AND_HIGHLIGHT))
+    dispatch(setActiveStepForSession(1))
+    dispatch(setExplanationSlidesCompletedForSession(true))
+    dispatch(setActivityIsCompleteForSession(false))
+    setHasStartedReadPassageStep(true)
+    setShowReadTheDirectionsButton(true)
+    setScrolledToEndOfPassage(false)
+    setStudentHighlights([])
+    setDoneHighlighting(false)
+    setHasStartedPromptsSteps(false)
+    setCompleteButtonClicked(false)
+  }
+
+  function preparePreviewBecauseStep() {
+    dispatch(setPreviewSessionStep(BECAUSE))
+    dispatch(setActiveStepForSession(2))
+    dispatch(setExplanationSlidesCompletedForSession(true))
+    dispatch(setActivityIsCompleteForSession(false))
+    setHasStartedReadPassageStep(true)
+    setShowReadTheDirectionsButton(false)
+    setScrolledToEndOfPassage(true)
+    setDoneHighlighting(true)
+    setHasStartedPromptsSteps(true)
+    setCompleteButtonClicked(false)
+    setCompletedSteps([1])
+  }
+
+  function preparePreviewButStep() {
+    dispatch(setPreviewSessionStep(BUT))
+    dispatch(setActiveStepForSession(3))
+    dispatch(setExplanationSlidesCompletedForSession(true))
+    dispatch(setActivityIsCompleteForSession(false))
+    setHasStartedReadPassageStep(true)
+    setShowReadTheDirectionsButton(false)
+    setScrolledToEndOfPassage(true)
+    setDoneHighlighting(true)
+    setHasStartedPromptsSteps(true)
+    setCompleteButtonClicked(false)
+    setCompletedSteps([1, 2])
+  }
+
+  function preparePreviewSoStep() {
+    dispatch(setPreviewSessionStep(SO))
+    dispatch(setActiveStepForSession(4))
+    dispatch(setExplanationSlidesCompletedForSession(true))
+    dispatch(setActivityIsCompleteForSession(false))
+    setHasStartedReadPassageStep(true)
+    setShowReadTheDirectionsButton(false)
+    setScrolledToEndOfPassage(true)
+    setDoneHighlighting(true)
+    setHasStartedPromptsSteps(true)
+    setCompleteButtonClicked(false)
+    setCompletedSteps([1, 2, 3])
+  }
+
+  React.useEffect(() => {
+    const { previewSessionStep } = session;
+
+    if(previewSessionStep === INTRODUCTION) {
+      preparePreviewIntroductionStep()
+    }
+    if(previewSessionStep === CHECKLIST) {
```

Comment:
I think line 297, 300, 303, 306 and 309 should all be `else if.`

Reply:
@brendanshean updated!